scripts/scripts.js

```diff
@@ -132,15 +133,228 @@ export function decorateMain(main) {
+            section = getSectionByElementSelector(prehidingSelector);
+          }
+          if (!section && cssSelector) {
+            section = getSectionByElementSelector(cssSelector);
+          }
+          if (section) {
+            section.classList.add('visibility-hidden');
+          }
+        }
+      });
+    }
+  });
+  document.body.classList.remove('visibility-hidden');
+  onDecoratedElement(async () => {
+    await window.alloy('applyPropositions', { propositions });
+
+    // keep track of propositions that were applied
+    propositions.forEach((p) => {
+      p.items = p.items.filter((i) => i.schema !== 'https://ns.adobe.com/personalization/dom-action' || !getElementForProposition(i));
+    });
+  });
+
+  // Reporting is deferred to avoid long tasks
+  window.setTimeout(() => {
+    // Report shown decisions
+    window.alloy('sendEvent', {
+      xdm: {
+        eventType: 'decisioning.propositionDisplay',
+        _experience: {
+          decisioning: { propositions },
+        },
+      },
+    });
+  });
+}
+
+async function sendAnalyticsEvent(xdmData) {
+  // eslint-disable-next-line no-undef
+  if (!alloy) {
+    return Promise.resolve();
+  }
+  // eslint-disable-next-line no-undef
+  return alloy('sendEvent', {
+    documentUnloading: true,
+    xdm: xdmData,
+  });
+}
+
+/**
+ * Basic tracking for page views with alloy
+ * @param document
+ * @param additionalXdmFields
+ * @returns {Promise<*>}
+ */
+// eslint-disable-next-line no-unused-vars
+async function analyticsTrackPageViews(document, additionalXdmFields = {}) {
+  const xdmData = {
+    eventType: 'web.webpagedetails.pageViews',
+    web: {
+      webPageDetails: {
+        pageViews: {
+          value: 1,
+        },
+        name: document.title,
+      },
+    },
+  };
+  sendAnalyticsEvent(xdmData);
+}
+
+function initializeTargetAnalytics() {
+  const alloyLoadedPromise = initWebSDK(`${window.hlx.codeBasePath}/scripts/alloy.min.js`);
+
+  if (getMetadata('target')) {
+    alloyLoadedPromise.then(() => {
+      fetch(`${getHostUrl()}/websdkconfig.json`)
+        .then((response) => response.json())
+        .then((configData) => {
+          // eslint-disable-next-line no-undef
+          alloy('configure', {
+            datastreamId: configData.data[0].DatastreamId,
+            orgId: configData.data[0].OrgId,
+          });
+          getAndApplyRenderDecisions();
+          analyticsTrackPageViews(document);
+        })
+        .catch((error) => {
+          // eslint-disable-next-line no-console
+          console.error('Error:', error);
+        });
+    });
+  }
+}
+
+/**
+ * Returns script that initializes a queue for each alloy instance,
+ * in order to be ready to receive events before the alloy library is loaded
+ * Documentation
+ * https://experienceleague.adobe.com/docs/experience-platform/edge/fundamentals/installing-the-sdk.html?lang=en#adding-the-code
+ * @type {string}
+ */
+function getAlloyInitScript() {
```

Comment:
I'd directly run the JS rather than wrapping this to a string and then an additional `<script>` tag. you are otherwise wasting CPU cycles and trigger DOM updates for no particular reason.

This is what was done in https://github.com/adobe-rnd/aem-martech/blob/main/src/index.js#L47-L59

Reply:
@ramboz  let me check this

Reply:
@ramboz updated